### ISSUE TITLE
B::Deparse doesn't deparse $# {$foo} correctly

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#12903]
Migrated from [rt.perl.org#117531](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=117531) (status was 'resolved')

Searchable as RT117531$

**Comment by p5pRT** [comment]
### From @nwc10

#### Created by @nwc10



$\#\{$foo\} and $\# \{$foo\} are parsed differently&#8203;:

$ perl \-le '\*\# = \{Bar => "Baz"\}; $foo = "Bar"; $\# = \\\*STDERR; print $\# \{$foo\}'
Baz

$ perl \-le '$foo = \[1\.\.3\]; $\# = \\\*STDERR; print $\#\{$foo\}'
$\# is no longer supported at \-e line 1\.
2


However\, B&#8203;::Deparse mishandles the former\, treating it the same as the latter&#8203;:

$ perl \-MO=Deparse \-le '\*\# = \{Bar => "Baz"\}; $foo = "Bar"; $\# = \\\*STDERR; print $\# \{$foo\}'
BEGIN \{ $/ = "\\n"; $\\ = "\\n"; \}
\*\# = \{'Bar'\, 'Baz'\};
$foo = 'Bar';
$\# = \\\*STDERR;
print $\#\{$foo\};
\-e syntax OK

\(note\, no space between $\# and \{\)

I noted this as severity "low"\. Arguably it's pretty much "wishlist"\.

Nicholas Clark

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=low
    module=B::Deparse

Site configuration information for perl 5.17.11:

Configured by nick at Fri Apr  5 12:01:22 CEST 2013.

Summary of my perl5 (revision 5 version 17 subversion 11) configuration:
  Derived from: 67de9eea8942d6ccebde223ac91854323ff60ebf
  Platform:
    osname=darwin, osvers=10.8.0, archname=darwin-2level
    uname='darwin mouse-mill.home 10.8.0 darwin kernel version 10.8.0: tue jun 7 16:33:36 pdt 2011; root:xnu-1504.15.3~1release_i386 i386 '
    config_args='-Dusedevel=y -Dcc=ccache clang-mp-3.2 -Dld=clang-mp-3.2 -Ubincompat5005 -Uinstallusrbinperl -Dcf_email=nick@ccl4.org -Dperladmin=nick@ccl4.org -Dinc_version_list=  -Dinc_version_list_init=0 -DDEBUGGING -Doptimize=-g -Uusethreads -Uuse64bitall -Uuselongdouble -Uusemymalloc -Duseperlio -Dprefix=/usr/local/blead -Dinstallman1dir=none -Dinstallman3dir=none -Dusevendorprefix -Dvendorprefix=/usr/local/blead/vendor -Uuserelocatableinc -Ud_dosuid -Uuseshrplib -de -Accccflags=-fcatch-undefined-behavior -Accccflags=-DNO_MATHOMS -Umad'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='ccache clang-mp-3.2', ccflags ='-fno-common -DPERL_DARWIN -no-cpp-precomp -DDEBUGGING -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/opt/local/include',
    optimize='-g',
    cppflags='-no-cpp-precomp -fno-common -DPERL_DARWIN -no-cpp-precomp -DDEBUGGING -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/opt/local/include'
    ccversion='', gccversion='4.2.1 Compatible Clang 3.2 (tags/RELEASE_32/final)', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='env MACOSX_DEPLOYMENT_TARGET=10.3 cc', ldflags =' -fstack-protector -L/usr/local/lib -L/opt/local/lib'
    libpth=/usr/local/lib /opt/local/lib /usr/lib
    libs=-lgdbm -ldbm -ldl -lm -lutil -lc
    perllibs=-ldl -lm -lutil -lc
    libc=, so=dylib, useshrplib=false, libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=bundle, d_dlsymun=undef, ccdlflags=' '
    cccdlflags=' ', lddlflags=' -bundle -undefined dynamic_lookup -L/usr/local/lib -L/opt/local/lib -fstack-protector'

Locally applied patches:
    


@INC for perl 5.17.11:
    lib
    /usr/local/blead/lib/perl5/site_perl/5.17.11/darwin-2level
    /usr/local/blead/lib/perl5/site_perl/5.17.11
    /usr/local/blead/vendor/lib/perl5/vendor_perl/5.17.11/darwin-2level
    /usr/local/blead/vendor/lib/perl5/vendor_perl/5.17.11
    /usr/local/blead/lib/perl5/5.17.11/darwin-2level
    /usr/local/blead/lib/perl5/5.17.11
    .


Environment for perl 5.17.11:
    DYLD_LIBRARY_PATH (unset)
    HOME=/Users/nick
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/opt/local/bin:/opt/local/sbin:/Users/nick/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/X11/bin:/usr/local/sbin:/sbin:/usr/sbin
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @cpansprout

Fixed in bcbe2b27bd9\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

Fixed in bcbe2b27bd9\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'new' to 'resolved'
